core/src/main/java/com/linecorp/armeria/server/EmptyContentDecodedHttpRequest.java

```diff
@@ -77,6 +77,11 @@ final class EmptyContentDecodedHttpRequest implements DecodedHttpRequest {
         this.exchangeType = exchangeType;
         this.requestStartTimeNanos = requestStartTimeNanos;
         this.requestStartTimeMicros = requestStartTimeMicros;
+        if (routingContext.sessionProtocol().isMultiplex()) {
+            serverMetrics.increaseActiveHttp2Requests();
+        } else {
+            serverMetrics.increaseActiveHttp1Requests();
```

Comment:
Could you elaborate on why we need to update the metrics in this class? Wouldn't it be better calling these in `Http[12]RequestDecoder`?

Reply:
I wanted to avoid making this change in the `serve0` method:
https://github.com/line/armeria/pull/5627/commits/0a77e61699001d88d39547376ce3513fae799fbe#diff-8c8fd516b9814fa10a52ccad8660a3383cc0862c1e0cc6f91b0a456eb11a3641R458-R476

However, I think we don't have any other choice but to make the change there due to the comment:
https://github.com/line/armeria/pull/5627#discussion_r1630468525

Reply:
I see. Let's wait for @seongbeenkim to address the comments, then. :wink:

Reply:
I pushed the commit. PTAL. 😉 
https://github.com/line/armeria/pull/5627/commits/41ee0f8dcb24050f7b25b2695cc3fb713c4668de

Reply:
Increasing pending requests in `Http[12]RequestDecoder`, decreasing pending requests and increasing active requests in `HttpServerHandler` look good to me.

This isn't just because of the comment https://github.com/line/armeria/pull/5627#discussion_r1630468525
but also because it simplifies understanding and managing the flow of metrics.

From my review, all classes implement `DecodedHttpRequest` are only created in `Http[12]RequestDecoder`.
if we manage the metrics within `Http[12]RequestDecoder` and `HttpServerHandler`, 
we could clearly observe that pending requests increase according to protocol in `Http[12]RequestDecoder` just before handling requests, and pending requests decreas and active requests increase in `HttpServerHandler`, regardless of the specific `DecodedHttpRequest` implementation.

However, if we manage the metrics within each implemented `DecodedHttpRequest`,
we couldn't see clearly when to decrease or increase pending and active requests because each class handles the metrics differently. This complexity makes it challenging to address issues when metrics need to be managed amidst the processing of requests, as noted in the comment.

Since all class implement DecodedHttpRequest are created through `Http[12]RequestDecoder` and are simply treated as DecodedHttpRequest in lots of classes including`HttpServerHandler`,
so i think handling the metrics directly through `DecodedHttpRequest` seems more straightforward than basing it on whether the requests are aggregated or streaming.

Reply:
That makes sense. Thanks for sharing your opinion. 👍